packages/page-signing/src/Verify.tsx

```diff
@@ -31,38 +33,18 @@ function Verify ({ className = '' }: Props): React.ReactElement {
     let cryptoType: CryptoTypes = 'unknown';
     let isValid = isValidPk && isValidSignature;
 
-    // We cannot just use the keyring verify since it may be an address. So here we first check
-    // for ed25519, if not valid, we try against sr25519 - if neither are valid, well, we have
-    // not been able to validate the signature
+    // We use signatureVerify to detect validity and crypto type
     if (isValid && publicKey) {
-      let isValidSr = false;
-      let isValidEd = false;
+      const verification = signatureVerify(data, signature, publicKey);
 
-      try {
-        isValidEd = naclVerify(data, signature, publicKey);
-      } catch (error) {
-        // do nothing, already set to false
-      }
-
-      if (isValidEd) {
-        cryptoType = 'ed25519';
-      } else {
-        try {
-          isValidSr = schnorrkelVerify(data, signature, publicKey);
-        } catch (error) {
-          // do nothing, already set to false
-        }
-
-        if (isValidSr) {
-          cryptoType = 'sr25519';
-        } else {
-          isValid = false;
-        }
+      if (verification.crypto !== 'none') {
+        isValid = verification.isValid;
+        cryptoType = verification.crypto;
       }
     }
 
     setValidity({ cryptoType, isValid });
-  }, [data, isValidPk, isValidSignature, publicKey, signature]);
+  }, [isEthereum, data, isValidPk, isValidSignature, publicKey, signature]);
```

Comment:
```suggestion
  }, [data, isValidPk, isValidSignature, publicKey, signature]);
```

Think this is not used now?